src/plugins/workspace/public/workspace_client.ts

```diff
@@ -0,0 +1,275 @@
+      const resultWithWritePermission = await this.list({
+        perPage: 999,
+      });
+      if (resultWithWritePermission?.success) {
+        const workspaceIdsWithWritePermission = resultWithWritePermission.result.workspaces.map(
+          (workspace: WorkspaceAttribute) => workspace.id
+        );
+        let workspaces = result.result.workspaces;
+        workspaces = result.result.workspaces.map((workspace: WorkspaceAttribute) => ({
+          ...workspace,
+          libraryReadonly: !workspaceIdsWithWritePermission.includes(workspace.id),
+        }));
+        this.workspaces.workspaceList$.next(workspaces);
+      }
+    }
+  }
+
+  public async enterWorkspace(id: string): Promise<IResponse<null>> {
+    const workspaceResp = await this.get(id);
+    if (workspaceResp.success) {
+      this.workspaces.currentWorkspaceId$.next(id);
+      return {
+        success: true,
+        result: null,
+      };
+    } else {
+      return workspaceResp;
+    }
+  }
+
+  public async getCurrentWorkspaceId(): Promise<IResponse<WorkspaceAttribute['id']>> {
+    const currentWorkspaceId = this.workspaces.currentWorkspaceId$.getValue();
+    if (!currentWorkspaceId) {
+      return {
+        success: false,
+        error: 'You are not in any workspace yet.',
+      };
+    }
+
+    return {
+      success: true,
+      result: currentWorkspaceId,
+    };
+  }
+
+  public async getCurrentWorkspace(): Promise<IResponse<WorkspaceAttribute>> {
+    const currentWorkspaceIdResp = await this.getCurrentWorkspaceId();
+    if (currentWorkspaceIdResp.success) {
+      const currentWorkspaceResp = await this.get(currentWorkspaceIdResp.result);
+      return currentWorkspaceResp;
+    } else {
+      return currentWorkspaceIdResp;
+    }
+  }
+
+  /**
+   * Persists an workspace
+   *
+   * @param attributes
+   * @returns
+   */
+  public async create(
+    attributes: Omit<WorkspaceAttribute, 'id'>
+  ): Promise<IResponse<WorkspaceAttribute>> {
+    const path = this.getPath([]);
+
+    const result = await this.safeFetch<WorkspaceAttribute>(path, {
+      method: 'POST',
+      body: JSON.stringify({
+        attributes,
+      }),
+    });
+
+    if (result.success) {
+      await this.updateWorkspaceList();
+    }
+
+    return result;
+  }
+
+  /**
+   * Deletes a workspace
+   *
+   * @param id
+   * @returns
+   */
+  public async delete(id: string): Promise<IResponse<null>> {
+    const result = await this.safeFetch<null>(this.getPath([id]), { method: 'DELETE' });
+
+    if (result.success) {
+      await this.updateWorkspaceList();
+    }
+
+    return result;
+  }
+
+  /**
+   * Search for workspaces
+   *
+   * @param {object} [options={}]
+   * @property {string} options.search
+   * @property {string} options.search_fields - see OpenSearch Simple Query String
+   *                                        Query field argument for more information
+   * @property {integer} [options.page=1]
+   * @property {integer} [options.per_page=20]
```

Comment:
The  type define in `WorkspaceFindOptions` is `perPage`, which one is more accurate?

Reply:
perPage is correct, fixed.

Reply:
It should have been fixed as well.